### PR TITLE
--overwrite will overwrite command package #2132

### PR DESCRIPTION
**Pull request by BrennanConroy**
#2132 
@victorhurdugaci

### CODE REVIEW

File: src/Microsoft.Framework.PackageManager/Install/InstallGlobalCommand.cs [hunk 1]
```diff
@@ -67,6 +67,17 @@ public FeedOptions FeedOptions
 
             var temporaryProjectFileFullPath = CreateTemporaryProject(FeedOptions.TargetPackagesFolder, packageId, packageVersion);
 
+            var packageFullPath = Path.Combine(
```
Comment (kichalla) [inline src/Microsoft.Framework.PackageManager/Install/InstallGlobalCommand.cs]: probably we should change the help text too? currently it says `Overwrites conflicting commands`

File: src/Microsoft.Framework.PackageManager/Install/InstallGlobalCommand.cs [hunk 2]
```diff
@@ -67,6 +67,17 @@ public FeedOptions FeedOptions
...
+                _commandsRepository.PackagesRoot.Root,
+                _commandsRepository.PathResolver.GetPackageDirectory(packageId, new SemanticVersion(packageVersion)));
+
+            if (OverwriteCommands)
+            {
+                WriteVerbose($"Deleting {packageFullPath}");
+                FileOperationUtils.DeleteFolder(packageFullPath);
+                Directory.Delete(packageFullPath);
```
Comment (troydai) [inline src/Microsoft.Framework.PackageManager/Install/InstallGlobalCommand.cs]: Need a meaningful error message when deleting fails.

**Comment by troydai** [comment]
:watch:

File: src/Microsoft.Framework.PackageManager/Install/InstallGlobalCommand.cs [hunk 3]
```diff
@@ -67,6 +67,17 @@ public FeedOptions FeedOptions
...
             var temporaryProjectFileFullPath = CreateTemporaryProject(FeedOptions.TargetPackagesFolder, packageId, packageVersion);
 
+            var packageFullPath = Path.Combine(
+                _commandsRepository.PackagesRoot.Root,
+                _commandsRepository.PathResolver.GetPackageDirectory(packageId, new SemanticVersion(packageVersion)));
+
+            if (OverwriteCommands)
+            {
+                WriteVerbose($"Deleting {packageFullPath}");
+                FileOperationUtils.DeleteFolder(packageFullPath);
```
Comment (victorhurdugaci) [inline src/Microsoft.Framework.PackageManager/Install/InstallGlobalCommand.cs]: Um... what happens if the packages is not installed? As in running with `--overwrite` when there is nothing to overwrite

Comment (BrennanConroy) [inline src/Microsoft.Framework.PackageManager/Install/InstallGlobalCommand.cs]: Ya... I fixed it, will update

File: src/Microsoft.Framework.PackageManager/Install/InstallGlobalCommand.cs (hunk 1 above)
Comment (BrennanConroy) [inline src/Microsoft.Framework.PackageManager/Install/InstallGlobalCommand.cs]: Suggestions?

**Comment by BrennanConroy** [comment]
Updated

File: src/Microsoft.Framework.PackageManager/Install/InstallGlobalCommand.cs [hunk 4]
```diff
@@ -67,6 +67,20 @@ public FeedOptions FeedOptions
...
 
+            var packageFullPath = Path.Combine(
+                _commandsRepository.PackagesRoot.Root,
+                _commandsRepository.PathResolver.GetPackageDirectory(packageId, new SemanticVersion(packageVersion)));
+
+            if (OverwriteCommands)
+            {
+                if (Directory.Exists(packageFullPath))
+                {
+                    WriteVerbose($"Deleting {packageFullPath}");
```
Comment (victorhurdugaci) [inline src/Microsoft.Framework.PackageManager/Install/InstallGlobalCommand.cs]: I think this should be WriteInformation. It is pretty important to know when things get deleted

**Comment by victorhurdugaci** [comment]
The change seems to be okay but I have one scenario that I would like you to test. I'm wondering if upgrade + overwrite works. Basically have package_v1 and packave_v1.1 and upgrade from package_v1 to v1.1 with overwrite.

I don't expect it to break but please check it. 

If it works, then :shipit:

File: src/Microsoft.Framework.PackageManager/Install/InstallGlobalCommand.cs [hunk 5]
```diff
@@ -67,6 +67,20 @@ public FeedOptions FeedOptions
...
+                    FileOperationUtils.DeleteFolder(packageFullPath);
+                    Directory.Delete(packageFullPath);
```
Comment (troydai) [inline src/Microsoft.Framework.PackageManager/Install/InstallGlobalCommand.cs]: Still doesn't capture the IOException if the deleting fails.

Comment (BrennanConroy) [inline src/Microsoft.Framework.PackageManager/Install/InstallGlobalCommand.cs]: I don't think we need to catch it. If it ever happens then we shouldn't continue the global install and the exception has as much information in it as we know.

**Pull request merged**